### PR TITLE
[SDPA-1213] Upgraded JSON:API 2.x and JSON:API Extras 3.x.

### PR DESCRIPTION
**Pull request by sonnykt**
JIRA: https://digital-engagement.atlassian.net/browse/SDPA-1213

Related PRs:
* Tide Core: https://github.com/dpc-sdp/tide_core/pull/57
* Tide Media: https://github.com/dpc-sdp/tide_media/pull/19
* Tide API: https://github.com/dpc-sdp/tide_api/pull/28
* Tide Site: https://github.com/dpc-sdp/tide_site/pull/18
* Tide Webform: https://github.com/dpc-sdp/tide_webform/pull/21
* Tide News: https://github.com/dpc-sdp/tide_news/pull/14
* Tide Page: https://github.com/dpc-sdp/tide_page/pull/14
* Tide Event: https://github.com/dpc-sdp/tide_event/pull/24
* Tide Landing Page: https://github.com/dpc-sdp/tide_landing_page/pull/63
* Tide Monsido: https://github.com/dpc-sdp/tide_monsido/pull/13
* Tide Search: https://github.com/dpc-sdp/tide_search/pull/11
* Tide: https://github.com/dpc-sdp/tide/pull/39
* Content-VIC: https://github.com/dpc-sdp/content-vic-gov-au/pull/462

### CODE REVIEW

File: src/Controller/TideApiController.php [hunk 1]
```diff
@@ -155,29 +155,35 @@ public static function create(ContainerInterface $container) {
...
+        ],
+      ],
+      'errors' => [$this->t('Path not found.')],
+    ];
+
     try {
       if ($path) {
         $cid = 'tide_api:route:path:' . hash('sha256', $path . $site);
 
+        $json_response['data']['id'] = $cid;
```
Comment (sonnykt) [inline src/Controller/TideApiController.php]: @enzolutions cache ID should not be exposed as `id`. Please use the returned entity UUID instead.

Comment (enzolutions) [reply src/Controller/TideApiController.php]: Thanks, @sonnykt for the suggestion I will implement the change

File: src/Controller/TideApiController.php [hunk 2]
```diff
@@ -223,11 +227,10 @@ public function getRoute(Request $request) {
...
 
             if ($code != Response::HTTP_BAD_REQUEST) {
-              $json_response['data'] = [
-                'status_code' => $redirect->getStatusCode(),
-                'type' => $type,
-                'redirect_url' => $redirect_url,
-              ];
+              $json_response['data']['status_code'] = $redirect->getStatusCode();
+              $json_response['data']['type'] = $type;
+              $json_response['data']['redirect_url'] = redirect_url;
```
Comment (sonnykt) [inline src/Controller/TideApiController.php]: @enzolutions `redirect_url` is an invalid symbol.

File: composer.json [hunk 3]
```diff
@@ -4,29 +4,23 @@
...
     "require": {
-        "drupal/jsonapi": "^1.24",
-        "drupal/jsonapi_extras": "^2.5",
+        "drupal/jsonapi": "^2.0",
+        "drupal/jsonapi_extras": "^3.3",
         "drupal/schemata": "^1.0-alpha2",
-        "drupal/openapi": "^1.0-alpha1",
-        "dpc-sdp/tide_core": "^1.0"
+        "dpc-sdp/tide_core": "dev-feature/SDPA-1213_metatag_upgrade as 1.0",
+        "dpc-sdp/tide_site": "dev-feature/SDPA-1213_jsonapi_upgrade as 1.0"
```
Comment (sonnykt) [inline composer.json]: @enzolutions @kurtfoster tide_site is not a dependency of tide_api.

**Comment by anthony-malkoun** [comment]
@sonnykt please review this. `build_suggest` isn't passing but that is because of dependency hell. The regular build seems to be working and debug suggests this is all working as intended.

File: composer.json (hunk 3 above)
Comment (kurtfoster) [reply composer.json]: @sonnykt , imho they shouldn't depend on each other. Unfortunately they have been coupled together now. I guess for now they need to depend on each other and there needs to be a ticket to decouple them.

**Pull request merged**